### ISSUE TITLE
`bundle install` failed with Rust 1.75

### ISSUE THREAD

**Issue by remore** [kateinoigakukun/ruby-compute-runtime#8]
I ran into the following error when I was running `bundle install` (as guided in the [README](https://github.com/kateinoigakukun/ruby-compute-runtime/blob/643c571b9681fee76afdd5c1926e4eccebb77e8e/README.md?plain=1#L15)) to follow the latest environment setup;

```
...
   Compiling wasmtime-wasi v9.0.4
   Compiling cap-std v0.24.4
   Compiling ambient-authority v0.0.1
error[E0061]: this method takes 1 argument but 2 arguments were supplied
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cap-primitives-0.24.4/src/rustix/linux/fs/set_times_impl.rs:20:25
   |
20 |             return file.set_times(
   |                         ^^^^^^^^^
21 |                 atime.map(SystemTimeSpec::into_std),
22 |                 mtime.map(SystemTimeSpec::into_std),
   |                 ----------------------------------- unexpected argument of type `Option<fs_set_times::SystemTimeSpec>`
   |
note: expected `FileTimes`, found `Option<SystemTimeSpec>`
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cap-primitives-0.24.4/src/rustix/linux/fs/set_times_impl.rs:21:17
   |
21 |                 atime.map(SystemTimeSpec::into_std),
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected struct `FileTimes`
                found enum `Option<fs_set_times::SystemTimeSpec>`
note: method defined here
  --> /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/fs.rs:695:12
help: remove the extra argument
   |
21 -                 atime.map(SystemTimeSpec::into_std),
21 +                 /* FileTimes */,
   |

error[E0061]: this method takes 1 argument but 2 arguments were supplied
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cap-primitives-0.24.4/src/rustix/linux/fs/set_times_impl.rs:34:25
   |
34 |             return file.set_times(
   |                         ^^^^^^^^^
35 |                 atime.map(SystemTimeSpec::into_std),
36 |                 mtime.map(SystemTimeSpec::into_std),
   |                 ----------------------------------- unexpected argument of type `Option<fs_set_times::SystemTimeSpec>`
   |
note: expected `FileTimes`, found `Option<SystemTimeSpec>`
  --> /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/cap-primitives-0.24.4/src/rustix/linux/fs/set_times_impl.rs:35:17
   |
35 |                 atime.map(SystemTimeSpec::into_std),
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: expected struct `FileTimes`
                found enum `Option<fs_set_times::SystemTimeSpec>`
note: method defined here
  --> /rustc/82e1608dfa6e0b5569232559e3d385fea5a93112/library/std/src/fs.rs:695:12
help: remove the extra argument
   |
35 -                 atime.map(SystemTimeSpec::into_std),
35 +                 /* FileTimes */,
   |

For more information about this error, try `rustc --explain E0061`.
error: could not compile `cap-primitives` (lib) due to 2 previous errors
warning: build failed, waiting for other jobs to finish...

cargo failed, exit code 101

Gem files will remain installed in /root/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/bundler/gems/ruby.wasm-f98b50c38e3d for inspection.
Results logged to /root/.rbenv/versions/3.3.0/lib/ruby/gems/3.3.0/bundler/gems/extensions/x86_64-linux/3.3.0/ruby.wasm-f98b50c38e3d/gem_make.out
...
```

Below is the version I used initially to set my environment up;
```
$ rustc --version
rustc 1.75.0 (82e1608df 2023-12-21)
$ rustup toolchain list
stable-x86_64-unknown-linux-gnu (default)
```

Temporarily I resolved this error by switching back to old version (in my case I went back to 1.68). I was not sure where to share this issue (maybe I should post this to ruby.wasm repo instead?) - happy to re-post this to ruby.wasm project if it helps.

**Comment by kateinoigakukun** [comment]
Oh, thank you for checking! I need to update some of crates used in wasi-vfs 👍

**Comment by kateinoigakukun** [comment]
Will be fixed by https://github.com/ruby/ruby.wasm/pull/364

**Comment by remore** [comment]
Awesome, thanks for updates and fixes as always! 🙏

**Comment by kateinoigakukun** [comment]
Closed by 36db89d547e5cc7a8f45778afab48ded7c5db183
